sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc

```diff
@@ -96,3 +96,77 @@ function dccr_ftr_workshop_countdown($action, $content_id) {
   $wrapper->save();
 }
 
+/**
+ * Implements hook_cron().
+ */
+function dccr_ftr_workshop_cron() {
+  $week_before = strtotime('-7 day',DCCR_CAMP_START_DATE);
+  $three_days_before = strtotime('-3 day',DCCR_CAMP_START_DATE);
+ if (DCCR_CAMP_START_DATE >= REQUEST_TIME && $week_before <= REQUEST_TIME && !variable_get('REMINDER_WEEK_BEFORE', FALSE)) {
+    variable_get('REMINDER_WEEK_BEFORE', TRUE);
+    dccr_ftr_workshop_invoke_reminde('7');
+  }
+  elseif (DCCR_CAMP_START_DATE >= REQUEST_TIME && $three_days_before <= REQUEST_TIME && !variable_get('REMINDER_THREE_DAYS_BEFORE', FALSE)) {
+    variable_set('REMINDER_THREE_DAYS_BEFORE', TRUE);
+    dccr_ftr_workshop_invoke_reminder('3');
+ }
+}
+
+/**
+ * Invoke the rule event dccr_ftr_workshop_nofications.
+ */
+function dccr_ftr_workshop_invoke_reminder($days) {
+  $result = db_select('node', 'n')
+    ->fields('n',array('nid'))
+    ->condition('type','workshop', '=')
+    ->condition('status',1, '=')
+    ->execute();
+  $workshops_nids = $result->fetchCol();
+  $workshops = node_load_multiple($workshops_nids);
+
+  foreach ($workshops_nids as $nid) {
```

Comment:
node_load_multiple returns an array keyed by nid; so I think you could iterate over $workshops `foreach ($workshops as $nid => $workshop)`